exchanges/okcoin/okcoin_websocket.go

```diff
@@ -147,150 +238,260 @@ func (o *OKCoin) WsHandleData(respRaw []byte) error {
 	var eventResponse WebsocketEventResponse
 	err = json.Unmarshal(respRaw, &eventResponse)
 	if err == nil && eventResponse.Event != "" {
-		if eventResponse.Event == "login" {
-			if o.Websocket.Match.Incoming("login") {
-				o.Websocket.SetCanUseAuthenticatedEndpoints(eventResponse.Success)
+		switch eventResponse.Event {
+		case "login":
+			if o.Websocket.Match.IncomingWithData("login", respRaw) {
+				o.Websocket.SetCanUseAuthenticatedEndpoints(eventResponse.Code == "0")
+			}
+		case "subscribe", "unsubscribe":
+			o.Websocket.DataHandler <- eventResponse
+		case "error":
+			if o.Verbose {
+				log.Debugf(log.ExchangeSys,
+					o.Name+" - "+eventResponse.Event+" on channel: "+eventResponse.Channel)
 			}
 		}
-		if o.Verbose {
-			log.Debugln(log.ExchangeSys,
-				o.Name+" - "+eventResponse.Event+" on channel: "+eventResponse.Channel)
-		}
 	}
 	return nil
 }
 
-// StringToOrderStatus converts order status IDs to internal types
-func StringToOrderStatus(num int64) (order.Status, error) {
-	switch num {
-	case -2:
-		return order.Rejected, nil
-	case -1:
-		return order.Cancelled, nil
-	case 0:
-		return order.Active, nil
-	case 1:
-		return order.PartiallyFilled, nil
-	case 2:
-		return order.Filled, nil
-	case 3:
-		return order.New, nil
-	case 4:
-		return order.PendingCancel, nil
-	default:
-		return order.UnknownStatus, fmt.Errorf("%v not recognised as order status", num)
+func (o *Okcoin) wsProcessAdvancedAlgoOrder(respRaw []byte) error {
+	var resp WebsocketAdvancedAlgoOrder
+	err := json.Unmarshal(respRaw, &resp)
+	if err != nil {
+		return err
+	}
+	o.Websocket.DataHandler <- resp
+	return nil
+}
+
+func (o *Okcoin) wsProcessAlgoOrder(respRaw []byte) error {
```

Comment:
Same here, please process response into an order.Detail to send to the websocketroutine_manager